src/printer.js

```diff
@@ -3831,6 +3845,7 @@ function nodeStr(node, options) {
   const alternate = preferred === single ? double : single;
 
   let shouldUseAlternateQuote = false;
+  let canChangeDirectiveQuotes = !rawContent.includes('\\');
```

Comment:
After a bit more thought, I decided to go ahead and make the logic a little more robust, so it covers the case you mentioned, about having an escaped `\` in the literal. See https://github.com/josephfrazier/prettier/commit/a52916f5755a8bc175d8fc519f318859cf3677ae

I also realized that the Github UI doesn't show any further changes on this branch past the ones that were merged, so I'll go ahead and open a new PR, linking back to this one.